### ISSUE TITLE
Large performance gap between test and val sets in intra-scan mode

### ISSUE THREAD

**Issue by BIT-SZZ** [soupeeli/NeuroBOLT#5]
Dear Authors, thank you for your excellent work and for sharing the code and data. I've been trying to reproduce the "intra-scan" results, following your documentation and using the provided data very strictly. I am running the following command: `python main.py --batch_size 16 --finetune ./checkpoints/labram-base.pth --labels_roi Thalamus --dataset VU --train_test_mode intrascan --dataname sub11-scan01` The only adjustment I made was to the dataset folder naming, as I noticed the original names seemed to have a slight mismatch with the paths expected by the data processing code. Everything else is exactly as per the repo.

I'm running into an unusual and consistent issue: the performance on the validation set is low (Correlation metrics around 0.2 - 0.3), while the performance on the test set is, in contrast, very high (Correlation metrics around 0.7). This large, counter-intuitive gap is puzzling. To investigate, I tried applying other models (both my own custom ones and other related open-source models) to this same data pipeline. Interestingly, I observed the exact same high-test / low-val discrepancy.

Could you please clarify if this behavior is expected for the intra-scan mode? I am concerned I might have missed a specific configuration step, or perhaps there's an issue in the data handling protocol I'm not seeing. Any clarification would be extremely helpful.

As a related question, I've also noticed that some models (including my attempts) tend to predict nearly identical outputs for the entire sequence, appearing as a flat line when visualized. My first thought was that the MSE Loss might be causing a regression to the mean. However, adding other loss functions hasn't resolved this. Did you encounter a similar "flat line" prediction problem during your model design, and do you have any insights on how to mitigate it?

Thanks for your time and any guidance you can offer!

**Comment by soupeeli** [comment]
Hi @BIT-SZZ , Thank you very much for your interest in our work! 

The behavior you observed can indeed occur in the intra-scan setting. In this scenario, the amount of training data is inherently limited, and both the validation and test splits are relatively short in temporal length (approximately 46–48 timepoints). With such short sequences, correlation-based metrics can be quite sensitive to noise. As a result, it is expected that performance may fluctuate across different splits, particularly if one segment contains higher noise levels or atypical signal dynamics. For some subjects or scans, it is also possible to observe cases where validation performance is much higher than test performance. Correlation tends to be more stable when evaluated on longer sequences or in inter-scan settings, where greater temporal coverage helps average out noise effects.

For the second question, I did not see flat-line predictions in the final trained models. In some cases, similar behavior can appear very early in training, but it typically disappears once the model starts to learn meaningful temporal patterns. I agree that adding additional loss like correlation can potentially help improve the performance but, in our experience, using MSE loss alone should work fine in this case. 

If you are using this repo and only replacing the model, I would suggest double-checking that the model architecutre (like if the input output shape is correct, activation function, etc..), and the data preprocessing & data loader as well if you are using other datasets. Or, if this is still unresolved, could you share a bit more detail about your setup - e.g., whether you are using intra-scan or inter-scan, and what changes (if any) were made compared to the original repository?
